src/runtime/rpc/rpc_endpoint.cc

```diff
@@ -691,11 +691,13 @@ void RPCEndpoint::Init() {
  *     the key to modify their behavior.
  */
 std::shared_ptr<RPCEndpoint> RPCEndpoint::Create(std::unique_ptr<RPCChannel> channel,
-                                                 std::string name, std::string remote_key) {
+                                                 std::string name, std::string remote_key,
+                                                 TypedPackedFunc<void()> fshutdown) {
```

Comment:
@tqchen this is needed when a Python-side session constructor may keep a reference to the underlying session. Does it make sense to you?

Reply:
TQ requests change to `fcleanup`